sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +916,227 @@ class RotatingCertificateCredential implements TokenCredential {
+
+Many credential implementations in the `@azure/identity` library have an underlying token cache which persists sensitive authentication data such as account information, access tokens, and refresh tokens. By default this data exists in an in memory cache which is specific to the credential instance. However, there are scenarios where an application needs persist it across executions in order to share the token cache across credentials. To accomplish this the `@azure/identity` provides the `tokenCachePersistenceOptions`.
+
+> IMPORTANT! The token cache contains sensitive data and MUST be protected to prevent compromising accounts. All application decisions regarding the persistence of the token cache must consider that a breach of its content will fully compromise all the accounts it contains.
+
+In `@azure/identity` starting from v2 we need to use the package `@azure/identity-cache-persistence` that provides an extension to the identity package to enable persistent token caching. The package `@azure/identity-cache-persistence` exports an extension object that you must pass as an argument to the top-level `useIdentityExtension` function from the `@azure/identity` package.
+
+```
+$ npm install --save @azure/identity-cache-persistence
+```
+
+Enable token cache persistence in your program as follows:
+
+```ts
+import { useIdentityExtension } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+```
+
+After calling `useIdentityExtension`, the persistent token cache extension is registered to the `@azure/identity` package and will be available on all credentials that support persistent token caching (those that have `tokenCachePersistenceOptions` in their constructor options).
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. The `@azure/identity` library will protect and cache sensitive token data using available platform data protection if `@azure/identity-cache-persistence` is configured. Otherwise it will use only in-memory caching.
+
+To configure a credential, such as the `InteractiveBrowserCredential`, to persist token data, simply set the `tokenCachePersistenceOptions` option. The simplest way to persist the token data for a credential is to to use the default `tokenCachePersistenceOptions`. This will persist and read token data from a shared persisted token cache protected to the current account.
```

Comment:
```suggestion
To configure a credential, such as the `InteractiveBrowserCredential`, to persist token data, set the `tokenCachePersistenceOptions` option. The simplest way to persist the token data for a credential is to use the default `tokenCachePersistenceOptions`. This will persist and read token data from a shared persisted token cache protected to the current account.
```